File: projects/Domain/Enum/Status.cs
using System.ComponentModel.DataAnnotations;
namespace Domain.Enum
{
	public static class Status
	{
		public const string Matching = "Disputando";
		public const string Scheduled = "Marcado";
		public const string Finish = "Encerrado";
		public const string Canceled = "Cancelado";
		public const string Eliminated = "Eliminado";
		public const string Classified = "Classificado";
		public const string Champion = "Campeão";
		public const string Created = "Criado";
	}
}

File: projects/Models/TS/Enum/status.ts
export enum Status {
	Matching,
	Scheduled,
	Finish,
	Canceled,
	Eliminated,
	Classified,
	Champion,
	Created
}
interface Matching {
	kind: Status.Matching;
	value: number;
	name: 'Disputando';
}

interface Scheduled {
	kind: Status.Scheduled;
	sideLength: number;
	name: 'Marcado';
}
interface Finish {
	kind: Status.Finish;
	sideLength: number;
	name: 'Encerrado';
}
interface Canceled {
	kind: Status.Canceled;
	sideLength: number;
	name: 'Cancelado';
}
interface Eliminated {
	kind: Status.Eliminated;
	sideLength: number;
	name: 'Eliminado';
}
interface Classified {
	kind: Status.Classified;
	sideLength: number;
	name: 'Classificado';
}
interface Champion {
	kind: Status.Champion;
	sideLength: number;
	name: 'Campeão';
}
interface Created {
	kind: Status.Created;
	sideLength: number;
	name: 'Criado';
}
